clang/test/Analysis/store-to-immutable-basic.cpp

```diff
@@ -0,0 +1,166 @@
+}
+
+// Write through a pointer to const memory
+void test_write_through_const_pointer() {
+  const int local_const = 10; // expected-note {{Memory region is in immutable space}}
+  int *ptr = (int*)&local_const;
+  *ptr = 20; // expected-warning {{Writing to immutable memory is undefined behavior}}
+  // expected-note@-1 {{Writing to immutable memory is undefined behavior. This memory region is marked as immutable and should not be modified}}
+}
+
+// Write to string literal (should be in immutable space)
+void test_write_to_string_literal() {
+  char *str = (char*)"hello";
+  str[0] = 'H'; // expected-warning {{Writing to immutable memory is undefined behavior}}
+  // expected-note@-1 {{Writing to immutable memory is undefined behavior. This memory region is marked as immutable and should not be modified}}
+}
+
+// Write to const array
+void test_write_to_const_array() {
+  const int arr[5] = {1, 2, 3, 4, 5};
+  int *ptr = (int*)arr;
+  ptr[0] = 10; // expected-warning {{Writing to immutable memory is undefined behavior}}
+  // expected-note@-1 {{Writing to immutable memory is undefined behavior. This memory region is marked as immutable and should not be modified}}
+}
+
+// Write to const struct member
+struct TestStruct {
+  const int x; // expected-note 2{{Memory region is in immutable space}}
+  int y;
+};
+
+void test_write_to_const_struct_member() {
+  TestStruct s = {1, 2};
+  int *ptr = (int*)&s.x;
+  *ptr = 10; // expected-warning {{Writing to immutable memory is undefined behavior}}
+  // expected-note@-1 {{Writing to immutable memory is undefined behavior. This memory region is marked as immutable and should not be modified}}
+}
+
+// Write to const global array
+const int global_array[3] = {1, 2, 3};
+
+void test_write_to_const_global_array() {
+  int *ptr = (int*)global_array;
+  ptr[0] = 10; // expected-warning {{Writing to immutable memory is undefined behavior}}
+  // expected-note@-1 {{Writing to immutable memory is undefined behavior. This memory region is marked as immutable and should not be modified}}
+}
+
+// Write to const global struct
+const TestStruct global_struct = {1, 2};
+
+void test_write_to_const_global_struct() {
+  int *ptr = (int*)&global_struct.x;
+  *ptr = 10; // expected-warning {{Writing to immutable memory is undefined behavior}}
+  // expected-note@-1 {{Writing to immutable memory is undefined behavior. This memory region is marked as immutable and should not be modified}}
+}
+
+// Write to const parameter
+void test_write_to_const_param(const int param) { // expected-note {{Memory region is in immutable space}}
+  *(int*)&param = 100; // expected-warning {{Writing to immutable memory is undefined behavior}}
+  // expected-note@-1 {{Writing to immutable memory is undefined behavior. This memory region is marked as immutable and should not be modified}}
+}
+
+// Write to const reference parameter
```

Comment:
This and similar comments seem redundant to the function names attached.
Maybe we could just drop the comment?

Reply:
Removed the redundant comments.